OpenRA.Game/MiniYaml.cs

```diff
@@ -215,64 +244,81 @@ static List<MiniYamlNode> FromLines(IEnumerable<string> lines, string filename,
 					// Leading and trailing whitespace is always trimmed from keys.
 					// Leading and trailing whitespace is trimmed from values unless they
 					// are marked with leading or trailing backslashes
-					var keyLength = line.Length - keyStart;
+					var keyLength = lineLength - keyStart;
 					var valueStart = -1;
 					var valueLength = 0;
 					var commentStart = -1;
-					for (var i = 0; i < line.Length; i++)
+					var hasEscapedHashChar = false;
+					var prevChar = '\0';
+					for (var i = keyStart; i < lineLength; i++)
 					{
-						if (valueStart < 0 && line[i] == ':')
+						var currChar = line[i];
+						if (currChar == '#')
+						{
+							if (i == 0 || prevChar != '\\')
```

Comment:
This'd probably need to be `i == keyStart`, but since `prevChar` is `\0` in that case I think we can just use
```suggestion
							if (prevChar != '\\')
```

Reply:
Yeah, `0` or `keyStart` it was meant to avoid array index being out of bounds. But yes it can be removed.